src/dotnet/commands/dotnet-remove/Program.cs

```diff
@@ -14,34 +14,50 @@
 
 namespace Microsoft.DotNet.Tools.Remove
 {
-    public class RemoveCommand : DispatchCommand
+    public class RemoveCommand
     {
-        protected override string HelpText => $@"{LocalizableStrings.NetRemoveCommand};
-
-{LocalizableStrings.Usage}: dotnet remove [options] <object> <command> [[--] <arg>...]]
-
-{LocalizableStrings.Options}:
-  -h|--help  {LocalizableStrings.HelpDefinition}
-
-{LocalizableStrings.Arguments}:
-  <object>   {LocalizableStrings.ArgumentsObjectDefinition}
-  <command>  {LocalizableStrings.ArgumentsCommandDefinition}
-
-Args:
-  {LocalizableStrings.ArgsDefinition}
-
-{LocalizableStrings.Commands}:
-  p2p        {LocalizableStrings.CommandP2PDefinition}";
-
-        protected override Dictionary<string, Func<string[], int>> BuiltInCommands => new Dictionary<string, Func<string[], int>>
+        private static List<Func<CommandLineApplication, CommandLineApplication>>
+            BuiltInCommands => new List<Func<CommandLineApplication, CommandLineApplication>>
         {
-            ["p2p"] = RemoveProjectToProjectReferenceCommand.Run,
+            RemoveProjectToProjectReferenceCommand.CreateApplication,
         };
 
         public static int Run(string[] args)
         {
-            var cmd = new RemoveCommand();
-            return cmd.Start(args);
+            DebugHelper.HandleDebugSwitch(ref args);
```

Comment:
This is the third time this pattern is repeated. Should we extract it and vary only what needs to be varied? Seems like the name, full name and builtincommands?